src/metric/populate_test.go

```diff
@@ -143,13 +155,39 @@ func TestPopulateK8s(t *testing.T) {
 	}
 
 	k8sVersion := &version.Info{GitVersion: "v1.15.42"}
-	err = p.Populate(foo, kubeletSpecs, i, "test-cluster", k8sVersion)
+	err = p.Populate(foo, kubeletSpecs, intgr, "test-cluster", k8sVersion)
 	require.IsType(t, err, data.PopulateResult{})
 	assert.Empty(t, err.(data.PopulateResult).Errors)
 
-	expectedInventory := sdk.Inventory{}
-	expectedInventory.SetItem("cluster", "name", expectedMetrics[0].Entity.Name)
-	expectedInventory.SetItem("cluster", "k8sVersion", k8sVersion.String())
-	expectedMetrics[0].Inventory = expectedInventory
-	assert.ElementsMatch(t, expectedMetrics, i.Data)
+	expectedInventory := inventory.New()
+	err = expectedInventory.SetItem("cluster", "name", expectedEntities[0].Metadata.Name)
+	require.NoError(t, err)
+
+	err = expectedInventory.SetItem("cluster", "k8sVersion", k8sVersion.String())
+	require.NoError(t, err)
+	expectedEntities[0].Inventory = expectedInventory
+
+	if len(expectedEntities) != len(intgr.Entities) {
+		t.Fatalf("missing required entities")
+	}
+
+	// Sort slices, so we can later diff them one-by-one for decent readability
+	entitySliceLesser := func(entities []*integration.Entity) func(i, j int) bool {
+		return func(i, j int) bool {
+			return strings.Compare(entities[i].Metadata.Name, entities[j].Metadata.Name) < 0
+		}
+	}
+
+	sort.Slice(intgr.Entities, entitySliceLesser(intgr.Entities))
+	sort.Slice(expectedEntities, entitySliceLesser(expectedEntities))
+
+	// Compare entities deeply, one by one, ignoring unexported fields
+	for j := range expectedEntities {
+		compareIgnoreFields := cmpopts.IgnoreUnexported(integration.Entity{}, metric.Set{}, inventory.Inventory{})
```

Comment:
This can be declared outside of the loop, right?

Reply:
Yup, no idea what that was doing inside